src/linker/Linker.Steps/DiscoverCustomOperatorsHandler.cs

```diff
@@ -133,7 +133,7 @@ TypeDefinition NonNullableType (TypeReference type)
 			// that we need to unwrap until we get to the Nullable<T>
 			while (!type.IsGenericInstance) {
 				Debug.Assert (type is TypeSpecification);
```

Comment:
Not sure this assert is providing much value over doing the cast and getting a classcastexception